From 4d31999ece2d5565f58c9cb929caa665f49bd26b Mon Sep 17 00:00:00 2001
From: Adam Ford <aford173@gmail.com>
Date: Thu, 25 Jun 2020 06:18:10 -0500
Subject: [PATCH 43/49] arm64: dts: beacon-imx8mm: Fix distorted Bluetooth
 Audio

The default UART speed is set to 115,200 which isn't enough bandwidth
to support 48KHz audio streaming, so the audio gets distorted.

The maximum speed of the Bluetooth UART interface is 4M baud, so
this patch increases the baud rate.

The Audio can be tested with the following:

pulseaudio --start --log-target=syslog

bluetoothctl
scan on
pair <Speaker Address>
connect <Speaker Address>
exit

speaker-test -D pulse -t wav -c2

Fixes: PHOENIX8-129

Signed-off-by: Adam Ford <aford173@gmail.com>
---
 arch/arm64/boot/dts/freescale/beacon-imx8mm-som.dtsi | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm64/boot/dts/freescale/beacon-imx8mm-som.dtsi b/arch/arm64/boot/dts/freescale/beacon-imx8mm-som.dtsi
index 7919347c693b..1c81eb005ede 100644
--- a/arch/arm64/boot/dts/freescale/beacon-imx8mm-som.dtsi
+++ b/arch/arm64/boot/dts/freescale/beacon-imx8mm-som.dtsi
@@ -221,6 +221,7 @@
 		shutdown-gpios = <&gpio2 6 GPIO_ACTIVE_HIGH>;
 		clocks = <&osc_32k>;
 		clock-names = "extclk";
+		max-speed = <4000000>;
 	};
 };
 
-- 
2.17.1

